agent/xds/rbac.go

```diff
@@ -188,11 +204,22 @@ func removePermissionPrecedence(perms []*rbacPermission, intentionDefaultAction
 	return out
 }
 
-func intentionToIntermediateRBACForm(ixn *structs.Intention, isHTTP bool) *rbacIntention {
+func intentionToIntermediateRBACForm(ixn *structs.Intention, isHTTP bool, bundle *pbpeering.PeeringTrustBundle) *rbacIntention {
 	rixn := &rbacIntention{
-		Source:     ixn.SourceServiceName(),
+		Source: rbacService{
+			ServiceName: ixn.SourceServiceName(),
+			peer:        ixn.SourcePeer,
+		},
 		Precedence: ixn.Precedence,
 	}
+
+	// imported services will have addition metadata used to override SpiffeID creation
+	// TODO(peering): is this true? Can there be a race condition where the bundle is not yet replicated before xds runs?
```

Comment:
You can require that the proxycfg have received the peer bundle here:
```
// Valid returns whether or not the snapshot has all required fields filled yet.
func (s *ConfigSnapshot) Valid() bool {
	switch s.Kind {
	case structs.ServiceKindConnectProxy:
		if s.Proxy.Mode == structs.ProxyModeTransparent && !s.ConnectProxy.MeshConfigSet {
			return false
		}
		return s.Roots != nil &&
			s.ConnectProxy.Leaf != nil &&
			s.ConnectProxy.IntentionsSet &&
			s.ConnectProxy.MeshConfigSet // <---- add something here
```
in agent/proxycfg/snapshot.go

Reply:
Just to clarify, do I need to call `Valid` in the codepaths? Or is the snapshot validated before it is sent to xds (and I just need to add an extra clause)

Reply:
Nvm I see it being called before the send

Reply:
Yeah that function is the gating function to even send a ConfigSnapshot _to_ the xds code.

Reply:
Hm just realized adding it to `Valid` stops config snapshot from sending in non-peered contexts. I may have to add validation to error out if peer != nil but there's no corresponding trust bundle in the map